### PR TITLE
Initial implementation of SQS broker

### PR DESCRIPTION
**Pull request by autodidaddict**
This is an initial implementation of an SQS broker plugin. This is actively being used and developed so I suspect more features/enhancements will be forthcoming.

### CODE REVIEW

**Comment by autodidaddict** [comment]
@asim not sure what's causing the build failure - looks like a Kubernetes registry issue unrelated to my plugin

**Comment by asim** [comment]
Question about the topics. Does it not make more sense to set the url as the broker address and then publish or subscribe to "1234/queue" or is this a standard thing when publishing and subscribing using SQS?

**Comment by autodidaddict** [comment]
Standard sqs. The only address you get is the full URL of the queue. Very
similar to the URL of an s3 bucket, etc

The broker itself has no address, since that's basically AWS proper. The
region is part of the environment variables

**Comment by asim** [comment]
So from a plugin abstraction standpoint, it does not make sense to publish or subscribe to a queue url. If you swap this out with any other message broker you're essentially publishing to a topic that's a url which is a bit strange. 

The more appropriate solution for go-micro would be setting the url as a broker address and then publishing/subscribing to "queue.fifo". The amazon sqs documentation also shows this as the example when creating, sending or receiving. The region/url would be a preset value/config. 

I don't think there's a scenario in which you would need to publish to multiple regions from the same broker instance. I've actually done that via the federation service https://github.com/micro/federation-srv.

Thoughts?

**Comment by autodidaddict** [comment]
You would then be forcing the consumer to split the queue URL on their own
because all Amazon gives you is the single queue URL, and the code would
then have to reassemble the URL anyway.

This would be okay but if Amazon changes it's URL composition then it would
break the plugin.

Just trying to avoid forcing developers into a manual or error prone
process. Note that the URL prefix is different per queue, not just per
account or IAM role. In other words, two queues don't share the same prefix
so you'd have to split before the ugly unique ID, making the queue name
"3848282938471/foo.fifo" this isn't much cleaner than the full URL IMHO

**Comment by autodidaddict** [comment]
I see where you're going with this.. From the AWS SDK Go examples:
```
resultURL, err := svc.GetQueueUrl(&sqs.GetQueueUrlInput{
        QueueName: aws.String(name),
    })
```

It's using the `GetQueueURL` function to extrapolate the URL. If I did it this way, there wouldn't be any need to supply a broker address, just supply the queue name and the service determines the full URL.

I'll mess with this internally and see how it looks and push new changes.

**Comment by asim** [comment]
Looks good. Only last thing. Can you please use [github.com/micro/go-log](https://github.com/micro/go-log) instead of fmt.Print for logging output. All libraries in the micro repo use it so the underlying logger can be swapped out. Basically just import that and swap fmt.Print for log.Log. Thanks.

**Comment by autodidaddict** [comment]
Woops. The println was a debug line I was planning to remove. I'll take
care of it

**Comment by asim** [comment]
Any update on this?

**Comment by asim** [comment]
Ping

**Comment by autodidaddict** [comment]
Really sorry about how long this took. I need a time machine to manufacture some spare time.

File: broker/sqs/sqs.go [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+
+	input := &sqs.SendMessageInput{
+		MessageBody: aws.String(string(msg.Body[:])),
+		QueueUrl:    &queueURL,
+	}
+	input.MessageAttributes = copyMessageHeader(msg)
+	input.MessageDeduplicationId = b.generateDedupID(msg)
+	input.MessageGroupId = b.generateGroupID(msg)
+
+	log.Log(fmt.Sprintf("Publishing SQS message, %d bytes", len(*msg.Body)))
```
Comment (asim) [inline broker/sqs/sqs.go]: It's complaining about this line. Should just be msg.Body rather than *msg.Body

Comment (autodidaddict) [reply broker/sqs/sqs.go]: Ugh. I hate having to do this by hand. I can't even use copy/paste. Will fix the typo

**Comment by asim** [comment]
Looks good. Thanks for this!

**Pull request merged**